machineconfiguration/v1/types_machineconfignode.go

```diff
@@ -0,0 +1,245 @@
+// or modifies in some way
+type MCOObjectReference struct {
+	// name is the name of the object being referenced. For example, this can represent a machine
+	// config pool or node name.
+	// Must be a lowercase RFC-1123 subdomain name (https://tools.ietf.org/html/rfc1123) consisting
+	// of only lowercase alphanumeric characters, hyphens (-), and periods (.), and must start and end
+	// with an alphanumeric character, and be at most 253 characters in length.
+	// +kubebuilder:validation:MaxLength:=253
+	// +kubebuilder:validation:XValidation:rule="!format.dns1123Subdomain().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+	// +required
+	Name string `json:"name"`
+}
+
+// MachineConfigNodeSpec describes the MachineConfigNode we are managing.
+type MachineConfigNodeSpec struct {
+	// node contains a reference to the node for this machine config node.
+	// +required
+	Node MCOObjectReference `json:"node"`
+
+	// pool contains a reference to the machine config pool that this machine config node's
+	// referenced node belongs to.
+	// +required
+	Pool MCOObjectReference `json:"pool"`
+
+	// configVersion holds the desired config version for the node targeted by this machine config node resource.
+	// The desired version represents the machine config the node will attempt to update to and gets set before the machine config operator validates
+	// the new machine config against the current machine config.
+	// +required
+	ConfigVersion MachineConfigNodeSpecMachineConfigVersion `json:"configVersion"`
+}
+
+// MachineConfigNodeStatus holds the reported information on a particular machine config node.
+type MachineConfigNodeStatus struct {
+	// conditions represent the observations of a machine config node's current state. Valid types are:
+	// UpdatePrepared, UpdateExecuted, UpdatePostActionComplete, UpdateComplete, Updated, Resumed,
+	// Drained, AppliedFilesAndOS, Cordoned, Uncordoned, RebootedNode, NodeDegraded, PinnedImageSetsProgressing,
+	// and PinnedImageSetsDegraded.
+	// +listType=map
+	// +listMapKey=type
+	// +kubebuilder:validation:MaxItems=20
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	// observedGeneration represents the generation of the MachineConfigNode object observed by the Machine Config Operator's controller.
+	// This field is updated when the controller observes a change to the desiredConfig in the configVersion of the machine config node spec.
+	// +kubebuilder:validation:XValidation:rule="self >= oldSelf", message="observedGeneration must not decrease"
+	// +kubebuilder:validation:Minimum=1
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+	// configVersion describes the current and desired machine config version for this node.
+	// +required
+	ConfigVersion MachineConfigNodeStatusMachineConfigVersion `json:"configVersion"`
```

Comment:
This struct has a required field within it, therefore should have a pointer when it is optional